src/components/modes/teacher/BackgroundToggle.js

```diff
@@ -25,26 +26,50 @@ const useStyles = makeStyles(() => ({
 const BackgroundToggle = () => {
   const classes = useStyles();
   const { t } = useTranslation();
-  const dispatch = useDispatch();
 
-  const { backgroundImage } = useSelector(
-    ({ appInstance }) => appInstance.content.settings,
+  const { mutate: postAppSetting } = useMutation(
+    MUTATION_KEYS.POST_APP_SETTING,
   );
 
-  const toggleDisabled = !backgroundImage?.uri;
+  const { mutate: patchAppSetting } = useMutation(
+    MUTATION_KEYS.PATCH_APP_SETTING,
+  );
+
+  const [ backgroundToggleSetting, setBackgroundToggleSetting ] = useState(null);
+  
+  const { data: appSettings, isSuccess } = useAppSettings();
+
+  useEffect(() => {
+    if(isSuccess) {
+      const backgroundSetting = appSettings?.find(
+        ({ name }) => name === APP_SETTINGS.BACKGROUND,
+      );
+      if(backgroundSetting){
+        setBackgroundToggleSetting(appSettings?.find(
+          ({ name }) => name === APP_SETTINGS.BACKGROUND_TOGGLE,
+        ));
+
+        if(backgroundToggleSetting === null) {
+          postAppSetting({
+            name: APP_SETTINGS.BACKGROUND_TOGGLE,
+            data: {
+              toggle: false,
+            }
+          });
+        }
```

Comment:
Why not have a default value? If it is null, it defaults to false. Once you toggle it you can create the setting (if id doesn't exist) or patch it.